packages/desktop/src/renderer/components/CreateUsername/CreateUsernameComponent.tsx

```diff
@@ -188,77 +231,137 @@ export const CreateUsernameComponent: React.FC<CreateUsernameComponentProps> = (
+                    <Typography variant='body2' className={classes.marginMedium}>
+                      We’re sorry, but the username <strong>{currentUsername && `@${currentUsername}`}</strong> was
+                      already claimed by someone else. <br />
+                      Can you choose another name?
+                    </Typography>
+
+                    <Typography variant='body2' className={classes.inputLabel}>
+                      Enter username
+                    </Typography>
+                  </>
                 )}
-              />
-              <div className={classes.gutter}>
-                {!errors.userName && userName.length > 0 && parsedNameDiffers && (
-                  <Grid container alignItems='center' direction='row'>
-                    <Grid item className={classes.iconDiv}>
-                      <WarningIcon className={classes.warrningIcon} />
-                    </Grid>
-                    <Grid item xs>
-                      <Typography
-                        variant='body2'
-                        className={classes.warrningMessage}
-                        data-testid={'createUserNameWarning'}
-                      >
-                        Your username will be registered as <b>{`@${userName}`}</b>
-                      </Typography>
-                    </Grid>
-                  </Grid>
+
+                <Controller
+                  control={control}
+                  defaultValue={''}
+                  rules={userFields.userName.validation}
+                  name={'userName'}
+                  render={({ field }) => (
+                    <TextInput
+                      {...userFields.userName.fieldProps}
+                      fullWidth
+                      classes={classNames({
+                        [classes.focus]: true,
+                        [classes.margin]: true,
+                        [classes.error]: errors.userName,
+                      })}
+                      placeholder={isNewUser ? 'Enter a username' : 'Username'}
+                      errors={errors}
+                      onPaste={e => e.preventDefault()}
+                      variant='outlined'
+                      onchange={event => {
+                        event.persist()
+                        const value = event.target.value
+                        onChange(value)
+                        // Call default
+                        field.onChange(event)
+                      }}
+                      onblur={() => {
+                        field.onBlur()
+                      }}
+                      value={field.value}
+                    />
+                  )}
+                />
+
+                {!isNewUser && (
+                  <Typography variant='caption' style={{ marginTop: 8 }}>
+                    Your username will be public, but you can choose any name you like. No spaces or special characters.
+                    Lowercase letters and numbers only.
```

Comment:
This validation should be added to certificatesRequests access controller if we'd like to enforce it.

Reply:
I talked with Bartek about general topic of Access Controllers and we think it's worth doing it as a separate task/PR and first thinking about all the places where to add them.

Reply:
I agree, I added some issues to track this:
- https://github.com/TryQuiet/quiet/issues/1893
- https://github.com/TryQuiet/quiet/issues/1899